src/parquet/column/reader.cc

```diff
@@ -165,6 +173,39 @@ bool TypedColumnReader<TYPE>::ReadNewPage() {
   return true;
 }
 
+// ----------------------------------------------------------------------
+// Batch read APIs
+
+static size_t DecodeMany(RleDecoder* decoder, int16_t* levels, size_t batch_size) {
+  size_t num_decoded = 0;
+
+  // TODO(wesm): Push this decoding down into RleDecoder itself
+  for (size_t i = 0; i < batch_size; ++i) {
+    if (!decoder->Get(levels + i)) {
```

Comment:
Right, only for EOS errors. I'm planning to write some tests that will explore these issues in some detail so we can define precisely what will / will not raise an exception.